src/main/java/seedu/dietmanager/logic/AppManager.java

```diff
@@ -72,8 +72,12 @@ private static void runApplication() {
                 String userInput = ui.readInput();
                 Optional<Command> command = CommandParser.parseInput(userInput);
                 if (command.isPresent()) {
-                    Result result = command.get().execute(profile, ui);
-                    ui.showMessage(result.showResult());
+                    if (isCommandValid(command.get())) {
+                        Result result = command.get().execute(profile, ui);
+                        ui.showMessage(result.showResult());
+                    } else {
+                        ui.displayCreateProfileMessage();
```

Comment:
I dont understand what your code does right now. 

First, you check isCommandValid, which return true when profile.isProfileExist() || command.getCommand().equals("set-profile") || command.getCommand().equals("help") || command.getCommand().equals("exit"). 

Next, when the user wants to exit but there is no profile created yet, you are going to propt the displayCreateProfileMessage which says: Please create a profile before using this command.

I don't think it is fair or even feasible to not let the user exit without creating a profile yet.

Reply:
ohhhh, i got it already, you trying to prompt the user to create a profile first prior to other command

Reply:
Erm its actually the opposite. I will restrict the commands to only set-profile, help and exit if the profile is not created yet since all our functions require a working profile to work.